### ISSUE TITLE
Vets-api terminal access for [Vanson Samuel]

### ISSUE THREAD

**Issue by binq** [department-of-veterans-affairs/va.gov-team#74212]
### Your Name

Vanson Samuel

### Your GitHub handle (terminal access is managed through GitHub)

@binq

### Team, Role, and Company of the target individual

VFS/VYE (Verify Your Enrollment) - Backend Development (Ruby on Rails) - GovCIO/Decisive Point Consulting Group

### Product Manager (PM) name and email

Darla van Nieukerk - Darla.vanNieukerk@va.gov

### Product Owner (PO) name and email

Jason Wolf - Jason.Wolf1@va.gov

### COR Name (leave "None" if you listed an AWS user name above)

Other - please specify in 'Additional Notes'

### Vendor Onboarding Representative (VOR) Name (leave "None" if you listed an AWS user name above)

GovernmentCIO - Kimberly O. West

### Please confirm if you are listed as a team member in Atlas or are listed in Platform Team Roster

- [x] [Team Member in Atlas](https://www.va.gov/atlas/product_directory/team_members)
- [X] [Platform Team Roster](https://vfs.atlassian.net/wiki/spaces/AP/pages/1908834623/Platform+Roster)

### Please confirm E-QIP Transmittal/Adjudication. This could be a screenshot of your Transmittal/Adjudication email, a link to another request where you've submitted this proof (i.e. AWS/SOCKS), etc.

- [X] I have attached proof that I have either transmitted my E-QIP survey or previously received a favorable decision (by including in "Additional Notes" section below)

### ArgoCD terminal access requested for the following environments

- [X] dev
- [X] staging
- [X] sandbox
- [x] production (if requesting, justify prod access in 'Additional Notes' section)

### Additional Notes

![image](https://github.com/department-of-veterans-affairs/va.gov-team/assets/37637/f22ec4d5-255d-4a1b-bbf0-86effe283d06)

COR: Joseph Riccio - [Joseph.Riccio@va.gov](mailto:Joseph.Riccio@va.gov) (edited)

### Additional resources for Support Engineer

- [X] For users on a VFS team but not listed as a team member in Atlas, they must start the [Platform orientation process](https://depo-platform-documentation.scrollhelp.site/getting-started/Platform-Orientation.1877344532.html)
- [X] Instructions for handling requests [here](https://vfs.atlassian.net/wiki/spaces/PTST/pages/2183037084/Backend+Support+Responsibilities#Handle-vets-api-ArgoCD-terminal-access) - Support Only

**Comment by bdech** [comment]
confirmed in Atlas.

**Comment by bdech** [comment]
@binq You’ve been granted access to the `dev`, `staging`, and `sandbox` environments.

Here's some additional documentation for [Vets API on EKS](https://depo-platform-documentation.scrollhelp.site/developer-docs/vets-api-on-eks#VetsAPIonEKS-TerminalAccess) that may help.

**Comment by binq** [comment]
I need prod access to launch the VYE project

**Comment by bmatos312** [comment]
@binq Hello Vanson, for prod acces this would require octo approval so we will have to wait until we here from them

**Comment by jsimonVA** [comment]
Hi @bmatos312  who can we talk to and try to nudge this along quicker. We are launching into prod today and its imperative @binq gets this access as soon as we can.

**Comment by bmatos312** [comment]
@jsimonVA I'll try to get an octo member on this but I can't make any promises as the process normally gets checked around once a week.

**Comment by jsimonVA** [comment]
@bmatos312  Do you have a name or a team we can reach out to?

**Comment by JonathanKamensDVA** [comment]
Typically we would like a bit more explanation than "we are launching into prod today" for why someone needs production access. What do you need to _do_ in prod? Wouldn't tasks related to deploying a new app into production typically be performed by the Platform team as part of the deployment?

I also need to understand better whether there is a need for long-term access to production, or whether it's just needed for the production deployment and can then be revoked.

Finally, on a process level, I question the wisdom of deploying a new app to production a few hours before a federal holiday, when we are already short-staffed today, we won't be working tomorrow, and we will be short-staffed on Friday because many people are taking days off surrounding the holiday. This does not seem like ideal timing.

**Comment by jsimonVA** [comment]
@binq can you respond to @JonathanKamensDVA questions. 

As for launch, we are launching today behind feature flags so that we can ingest files that we need and set our selves up for a go live date next week. The site will not be visible/accessible to the public until next week.

**Comment by bmatos312** [comment]
@jsimonVA @binq Is the functionality veteran impacting requiring the expedited service as well, Just trying to see what this affects and impact level.

**Comment by binq** [comment]
Yes the functionality requiring the expedited service is veteran impacting

**Comment by bmatos312** [comment]
@andreahewitt-odd Tagging for awareness on this issue as well

**Comment by binq** [comment]
> Typically we would like a bit more explanation than "we are launching into prod today" for why someone needs production access. What do you need to _do_ in prod? Wouldn't tasks related to deploying a new app into production typically be performed by the Platform team as part of the deployment?

I need to run a one time rake task to populate verification history form the old app on the day we start letting users in. We are 'launching' today but we are still behind a feature flag.

> 
> I also need to understand better whether there is a need for long-term access to production, or whether it's just needed for the production deployment and can then be revoked.

It's just needed for next week, and can then be revoked.

> 
> Finally, on a process level, I question the wisdom of deploying a new app to production a few hours before a federal holiday, when we are already short-staffed today, we won't be working tomorrow, and we will be short-staffed on Friday because many people are taking days off surrounding the holiday. This does not seem like ideal timing.

We'll be behind a feature flag. So wouldn't call it launching. That's for next week.

**Comment by JonathanKamensDVA** [comment]
Thank you for the clarification.

Access approved for one month.

**Comment by jsimonVA** [comment]
Thank you all for the help with this. Much appreciated.

**Comment by binq** [comment]
> Thank you for the clarification.
> 
> Access approved for one month.

Thank You

**Comment by lindsay-insco** [comment]
@binq Your Production access has expired.

Would you like to request an extension? You have 72 hours to respond, or your
access will be revoked and you will need to fill out a new request

**Comment by binq** [comment]
Yes please.

**Comment by bdech** [comment]
Hey @binq, I see originally your access was approved for one month. Can you provide a brief explanation, such as some responsibilities that will require production access?

**Comment by bdech** [comment]
@binq bump

**Comment by binq** [comment]
Responsibilities that will require further production access include:
- We have yet to fully complete the migration from the legacy WAVE application to VA.gov. Once it is done some manual migration activity will be necessary.
- We are having production issues and being able to access production data aides in debugging.
- We are manually kicking of ingess/egress jobs as we were in the process of confirming how reliable our upstream services are. This has been completed however the pull request for setting up periodic jobs have yet to be approved: https://github.com/department-of-veterans-affairs/vets-api/pull/18432
- Once the ingess/egress jobs are scheduled to run periodically it will be necessary to delete files from S3 buckets to prevent ingress with stale data should our upstream partners fail to upload files as expected. (We are working on jobs that can be scheduled to handle this.)

**Comment by bdech** [comment]
@JonathanKamensDVA, Can you please review this extension for Vets-API prod access? Originally the access was approved for one month.

**Comment by JonathanKamensDVA** [comment]
Approved.

**Comment by bdech** [comment]
@binq, your Vets-api Production access has been extended for 6 months, expiring 03/16/25.

Please set a reminder for yourself to return to this Issue and request an extension within 1 week of the expiration date. You'll 
receive 1 reminder with a 48 hour window to respond. Failure to respond will result in your access being revoked and will require 
you to fill out a new request.

**Comment by bdech** [comment]
@binq  your Vets API Terminal Production access expires on 03/16/2025

Would you like to request an extension? You have 72 hours to respond, or your access will be revoked and you will need to fill out a new request.

**Comment by binq** [comment]
Yes, as tech lead working on VYE, EDM, and VEBT, I'd like to continue having access to vets-api terminal to deal with issues in production.

**Comment by bdech** [comment]
Support slack reminder [here](https://dsva.slack.com/archives/C06CY71PCAX/p1742318115682869)
